app/controllers/versions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class VersionsController < ApplicationController
+  respond_to :html, :json
+
+  def show
+    @article = Article.find_by(slug: params[:article_id]).decorate
+
+    @current_version = @article.versions.find_by(id: params[:id])
+    @previous_version = @current_version.previous
+
+    if @previous_version && @previous_version.object.present?
```

Comment:
And, what do you think about this refactor?

```ruby
    if @previous_version
      previous_content = if @previous_version.object.present?
                           @previous_version.reify.content
                         else
                           YAML.load(@previous_version.object_changes)["content"].last
                         end
      @diff = Diffy::Diff.new(previous_content, @current_version.item.content).to_s
    end
```
That way, we avoid similar calls